dipy/tracking/local_tracking.py

```diff
@@ -122,45 +155,60 @@ def _generate_tractogram(self):
 
         F = np.empty((self.max_length + 1, 3), dtype=float)
         B = F.copy()
-        for s in self.seeds:
+        for i, s in enumerate(self.seeds):
             s = np.dot(lin, s) + offset
-            # Set the random seed in numpy, random and fast_numpy (libc.stdlib)
+            # Set the random seed in numpy, random and fast_numpy (lic.stdlib)
             if self.random_seed is not None:
                 s_random_seed = hash(np.abs((np.sum(s)) + self.random_seed)) \
                     % (2**32 - 1)
                 random.seed(s_random_seed)
                 np.random.seed(s_random_seed)
                 fast_numpy.seed(s_random_seed)
-            directions = self.direction_getter.initial_direction(s)
-            if directions.size == 0 and self.return_all:
+
+            if self.initial_directions is None:
+                directions = self.direction_getter.initial_direction(s)
+            else:
+                directions = []
+                for d in self.initial_directions[i, :, :]:
+                    d_norm = np.linalg.norm(d)
+                    if d_norm > 0:
+                        new_d = d / d_norm
+                        directions.append(new_d)
+                directions = np.array(directions)
```

Comment:
This can be simplify, I am not even sure you need a loop, maybe you can do it all with numpy.

Reply:
done. thanks.